src/main/groovy/net/neoforged/gradleutils/tasks/CIConfigExtractionTask.java

```diff
@@ -125,4 +130,40 @@ private String exportResource() throws Exception {
 
         return jarFolder + templateZipName;
     }
+
+    private boolean disableDeletion;
+
+    @Input
+    public boolean getDisableDeletion() {
+        return this.disableDeletion;
+    }
+
+    @Option(option = "disable-deletion", description = "Disable deletion of existing workflows")
+    public void setDisableDeletion(boolean disableDeletion) {
+        this.disableDeletion = disableDeletion;
+    }
+
+    /**
+     * Finds the most common 0-based substring of all {@code groups}.
+     */
+    protected static String findCommonSubstring(List<String> groups) {
+        if (groups.isEmpty()) {
+            return "";
+        }
+
+        final Deque<Character> firstGroup = new ArrayDeque<>();
+        groups.get(0).chars().forEach(ch -> firstGroup.add(Character.valueOf((char) ch)));
```

Comment:
I wonder if it's worth determining the shortest string in the `groups` list, and then using that as the basis for finding the common prefix (on the basis that the longest common prefix possible is bounded by the size of the shortest string in the set).